src/Symfony/Component/Uid/Attribute/AsUid.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid\Attribute;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AsUid
+{
+    public const FORMAT_CANONICAL = 'canonical';
+    public const FORMAT_BASE58 = 'base58';
+    public const FORMAT_BASE32 = 'base32';
+    public const FORMAT_RFC4122 = 'rfc4122';
+    public const FORMAT_ANY = 'any';
+
+    public function __construct(public ?string $format = null)
```

Comment:
Same question: why nullable and not mandatory?

Reply:
I'll remove the attribute.